src/lib/stats.js

```diff
@@ -47,6 +47,11 @@ exports.mean = function(data, len) {
     return exports.aggNums(function(a, b) { return a + b; }, 0, data) / len;
 };
 
+exports.geoMean = function(data, len) {
+    if(!len) len = exports.len(data);
+    return exports.aggNums(function(a, b) { return a * b; }, 0, data) ** (1 / len);
```

Comment:
Does this actually work? I'd think the `0` (the initialization value for the aggregation) needs to be a `1` for geometric mean. Also you probably need to include whatever we name this in the [regex that separates the aggregator from the order](https://github.com/plotly/plotly.js/blob/a10b927bd64259d029578e3f5948ded2275c2ac5/src/plots/plots.js#L3095)

Reply:
On another note `**` operator is not working on IE according to the following:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Exponentiation

Perhaps better to replace it `Math.pow`?

Reply:
> I'd think the 0 (the initialization value for the aggregation) needs to be a 1 for geometric mean.
include whatever we name this in the [regex that separates the aggregator from the order](https://github.com/plotly/plotly.js/blob/a10b927bd64259d029578e3f5948ded2275c2ac5/src/plots/plots.js#L3095)

> ** operator is not working on IE

Great catches
resolved